### PR TITLE
Add blacklist to _RPTEventBroadcast to avoid infinite loop.

### PR DESCRIPTION
**Pull request by vito1188**

### CODE REVIEW

File: RPerformanceTracking/Private/NSURL+RPerformanceTracking.m [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+{
+    static NSURL *RATEndPointURL;
+    static dispatch_once_t once;
+    dispatch_once(&once, ^{
+        NSString *RATEndPoint = [NSBundle.mainBundle objectForInfoDictionaryKey:@"RATEndpoint"];
+        NSURL *url = [NSURL URLWithString:RATEndPoint];
+        RATEndPointURL = url ?: [NSURL URLWithString:@"https://rat.rakuten.co.jp/"];
+    });
+
+    if (!self || [self.host isEqualToString:RATEndPointURL.host]) {
```
Comment (donnie-jp) [inline RPerformanceTracking/Private/NSURL+RPerformanceTracking.m]: `self.host` will check for nil self so the extra `!self` check is redundant

File: RPerformanceTracking/Private/_RPTTracker.m [hunk 2]
```diff
@@ -127,8 +128,8 @@ - (void)updateURL:(NSURL *)url trackingIdentifier:(uint_fast64_t)trackingIdentif
 - (void)sendResponseHeaders:(NSDictionary *)responseHeaders trackingIdentifier:(uint_fast64_t)trackingIdentifier
 {
     _RPTMeasurement *measurement = [_ringBuffer measurementWithTrackingIdentifier:trackingIdentifier];
-    NSString *sourceURL = (NSString *)measurement.receiver;
-    if (!sourceURL.length) {
+    NSString *sourceURLString = (NSString *)measurement.receiver;
+    if ([[NSURL URLWithString:sourceURLString] isBlacklisted]) {
```
Comment (donnie-jp) [inline RPerformanceTracking/Private/_RPTTracker.m]: I think it should still check for valid length before calling isBlacklisted

**Review by donnie-jp — CHANGES_REQUESTED** [review]
comments inline

File: Tests/TrackerTests.m [hunk 3]
```diff
@@ -118,6 +119,116 @@
...
+
+            it(@"should call 'sendEventName:topLevelDataObject:' method with the topLevelDataObject.perfData.entries having one item which have 'name' is equal to the url of measurement", ^{
+                KWCaptureSpy *spy = [_RPTEventBroadcast captureArgument:@selector(sendEventName:topLevelDataObject:) atIndex:1];
+
+                [tracker sendResponseHeaders:@{@"foo": @"bar"} trackingIdentifier:100];
+
+                NSDictionary *dict = spy.argument;
+                NSArray *entries = dict[@"perfdata"][@"entries"];
+                NSDictionary *entryData = entries.firstObject;
+                [[entryData[@"name"] should] equal:@"https://google.com"];
```
Comment (donnie-jp) [inline Tests/TrackerTests.m]: Why is the url `https://google.com`? I know why, but it isn't obvious from reading the test, so I think setting the `measurement.receiver` should be in the actual test not the beforeEach.

**Review by donnie-jp — APPROVED** [review]

**Comment by donnie-jp** [comment]
LGTM
The small comments can be addressed now or later.

File: Tests/TrackerTests.m (hunk 3 above)
Comment (vito1188) [reply Tests/TrackerTests.m]: I will update it.

**Review by donnie-jp — APPROVED** [review]
Approved

**Pull request merged**